docs/building-apps/building-static-apps.rst

```diff
@@ -83,15 +83,9 @@ sub-optimal:
 
    In other words: *using dynamic linking saves memory.*
 
-#. If you disable Open MPI's ``dlopen`` functionality (which is
-   necessary to create *fully*-static MPI applications), you lose the
-   following:
-
-   * CUDA, because |mdash| among other reasons |mdash| the CUDA
-     library is dynamically loaded at run-time via ``dlopen(3)``.
-
-   * Memory manager functionality, which is important for OS-bypass
-     networks such as InfiniBand.
+#. *Fully*-static applications lose certain Open MPI functionalities
+   that depend on ``dlopen``, an example being memory manager which is
+   important for OS-bypass networks such as InfiniBand.
```

Comment:
I honestly cannot remember why this is a true statement.  Does the Open MPI memory manager use `dlsym(3)` or something?

Reply:
Hmmmm a quick search found 2 places:
-  https://github.com/open-mpi/ompi/blob/1fdd32db901681a8727440ccd037710ed8da1228/opal/mca/memory/malloc_solaris/memory_malloc_solaris_component.c#L32
- https://github.com/wenduwan/ompi/blob/970d0dc2da878b2215eccfec6eca6dc50e6a35e6/opal/mca/memory/patcher/memory_patcher_component.c#L37

The only use of `dlsym` is in `opal/mca/memory` https://github.com/wenduwan/ompi/blob/970d0dc2da878b2215eccfec6eca6dc50e6a35e6/opal/mca/memory/malloc_solaris/memory_malloc_solaris_component.c#L132

... but that is optional and NOT preferred


https://github.com/wenduwan/ompi/blob/970d0dc2da878b2215eccfec6eca6dc50e6a35e6/opal/mca/memory/malloc_solaris/configure.m4#L67-L79

Reply:
After some discussion off-list, here's an observation from @bwbarrett:

> At one point (I haven’t looked at `configure` recently), `--disable-dlopen` meant that we wouldn’t link against `libdl` and therefore `dlsym` wouldn’t work.  Further, patcher requires that libc be a dynamic library, so if you did fully static link the MPI application, that `dlsym` would fail.  So that statement is true.
>  
> However, if you specify `–enable-mca-static`, which just makes all the components build into `libmpi` and friends, then patcher will happily work.